addon/components/sl-modal.js

```diff
@@ -0,0 +1,236 @@
+    // Properties
+
+
+    /**
+     * aria-describedby attribute
+     *
+     * @type {?String}
+     * */
+    'aria-describedby': null,
+
+    /**
+     *  aria-hidden attribute to inform assistive technologies to skip the
+     *  modal's DOM elements
+     *
+     * @type {String}
+     *                          */
+    'aria-hidden': 'true',
+
+    /**
+     * The aria-role value
+     *
+     * @type {String}
+     *                     */
+    ariaRole: 'dialog',
+
+    /*
+     * Whether to show Bootstrap's backdrop
+     *
+     * @type {Boolean|String}
+     */
+    backdrop: true,
+
+    /*
+     * Whether to modal is open or not
+     *
+     * @type {Boolean}
+     */
+    isOpen: false,
+
+    /**
+     * The modal service used to register and unregister this modal component
+     *
+     * @type {ember/Service}
+     */
+    modalService: Ember.inject.service( 'sl-modal' ),
+
+   /**
+    * The unique name use to register this modal with the modal service
+    *
+    * If this value is left null(default), then the component will not
+    * be registered on the modal service.
+    *
+    * @type {?String}
+    */
+    name: null,
+
+    /**
+     * tabindex attribute value
+     *
+     * @type {String}
+     */
+    tabindex: '-1',
+
+
+    // -----------------------------------------------------
+    // Observers
+
+
+    /**
+     * aria-labelledby attribute value
+     *
+     * @function
+     * @returns {String}
+     */
+    'aria-labelledby': Ember.computed(
+        'elementId',
+        function() {
+            return 'modalTitle' + this.get( 'elementId' );
+        }
+    ),
+
+    /**
+     * Register component on the modal service
+     *
+     * @function
+     * @returns {function}
+     */
+    register: Ember.on(
+       'init',
+       function() {
+           let name = this.get( 'name' );
+
+           if ( name ) {
+              this.get( 'modalService' ).register( this, name );
+           }
+       }
+    ),
+
+    /**
+     * Set up the component as a Bootstrap Modal and listen for events
+     *
+     * @function
+     * @returns {function}
+     */
+     setupModal: Ember.on(
+         'didInsertElement',
+         function() {
+            let modal = this.$().modal({
+                keyboard: true,
+                show: false,
+                backdrop: this.get( 'backdrop' )
+            });
+
+            modal.on( 'show.bs.modal', () => {
+               this.sendAction( 'beforeShow' );
+            });
+
+            modal.on( 'shown.bs.modal', () => {
+               this.set( 'isOpen', true );
+               this.sendAction( 'afterShow' );
+            });
+
+            modal.on( 'hide.bs.modal', () => {
+               this.sendAction( 'beforeHide' );
+            });
+
+            modal.on( 'hidden.bs.modal', () => {
+               this.set( 'isOpen', false );
+               this.sendAction( 'afterHide' );
+            });
+         }
```

Comment:
The `remote` property is deprecated since v3.3.0 of bootstrap, i'm not sure we should support the loaded event when the property that is related to this event is no longer around.